src/App.tsx

```diff
@@ -1,156 +1,83 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect, useMemo } from 'react';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+import { ErrorType } from './types/ErrorType';
+
+const USER_ID = 6373;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterByStatus, setFilterByStatus] = useState(FilterStatus.All);
+  const [errorType, setErrorType] = useState(ErrorType.None);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      try {
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch (error) {
+        setHasError(true);
+        setErrorType(ErrorType.Load);
+      }
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed).length;
+
+  const filterTodos = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterByStatus) {
+        case FilterStatus.All:
```

Comment:
as for me this case is default, in case of everything have appropriate and correct types